test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java

```diff
@@ -385,6 +391,13 @@ public void close() throws IOException {
         }
     }
 
+    private void propagateApplicationConfig(AppArtifact artifact, Map<String, String> sysPropRestore) {
+        sysPropRestore.put(QUARKUS_APPLICATION_NAME, System.getProperty(QUARKUS_APPLICATION_NAME));
```

Comment:
Just a small hint: System.setProperty() returns the previous property.

Reply:
nice!